### DISCUSSION TITLE
Cross platform lockfile generation issues

### DISCUSSION THREAD

**Discussion by ShantanuKumar** [pdm-project/pdm#1930] [Q&A]
* I see that even after running `pdm lock -G:all`, `pdm lock --check` constantly keeps telling me that the lockfile doesn't match pyproject.toml. I don't know if it's because I imported an existing `pyproject.toml` which had `flit` as a build system. But I am unable to fix this.

```sh
pdm lock -G:all
🔒 Lock successful
Changes are written to pdm.lock.

# git diff shows no changes

pdm lock --check -v
Lock file hash doesn't match pyproject.toml, packages may be outdated
✖ Lockfile is out of date.
```

* The other issue I have seen is that lockfiles most probably aren't accounting for all platform/architecture and their transitive dependencies correctly. I am working inside a docker container on Mac where I generate the lockfile. The CI/CD runs inside docker container on gitlab and always wants to update the lockfile. This is what I constantly see in our CI. So this is not just specific to one package.
```sh
STATUS: Resolving: new pin pyright 1.1.299
STATUS: Resolving: new pin tensorflow 2.11.0
STATUS: Resolving: new pin numpy 1.24.3
STATUS: Resolving: new pin protobuf 3.19.6
STATUS: Resolving: new pin dash 2.8.1
STATUS: Resolving: new pin matplotlib 3.7.1
STATUS: Resolving: new pin pillow 9.5.0
STATUS: Resolving: new pin pandas 2.0.1
STATUS: Resolving: new pin arrow 1.2.3
STATUS: Resolving: new pin black 23.3.0
STATUS: Resolving: new pin boto3 1.26.134
STATUS: Resolving: new pin boto3-stubs 1.26.134
STATUS: Resolving: new pin cryptography 40.0.2
```

**Comment by pawamoy** [comment]
I think this is solved by https://github.com/pdm-project/pdm/commit/c2888809a66ef2bd93c8ddabece452f8f21fd0c8.
